tests/test_pandas.py

```diff
@@ -353,22 +345,23 @@ def write_to_db():
             method=make_pd_writer(quote_identifiers=quote_identifiers),
         )
 
-    try:
-        if quote_identifiers:
-            with pytest.raises(
-                ProgrammingError,
-                match=r".*SQL compilation error.*\ninvalid identifier '\"a\"'.*",
-            ):
+    with engine_testaccount.connect() as conn:
+        try:
+            if quote_identifiers:
+                with pytest.raises(
+                    ProgrammingError,
+                    match=r".*SQL compilation error.*\ninvalid identifier '\"a\"'.*",
+                ):
+                    write_to_db()
+            else:
                 write_to_db()
-        else:
-            write_to_db()
-            results = sorted(
-                conn.exec_driver_sql(f"SELECT * FROM {table_name}").fetchall(),
-                key=lambda x: x[0],
-            )
-            # Verify that all 10 entries were written to the DB
-            for i in range(10):
-                assert results[i] == (i, i + 10)
-            assert len(results) == 10
-    finally:
-        conn.exec_driver_sql(f"DROP TABLE IF EXISTS {table_name}")
+                results = sorted(
+                    conn.exec_driver_sql(f"SELECT * FROM {table_name}").fetchall(),
+                    key=lambda x: x[0],
```

Comment:
You could use [itemgetter](https://docs.python.org/3/library/operator.html#operator.itemgetter) here